### PR TITLE
feat: Adding Volcano Batch Scheduler Helm Chart 

### PR DESCRIPTION
**Pull request by elygerou**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/main/.github/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Added helm chart for volcano
### Motivation

- Resolves #304 from repo: [data-on-eks](https://github.com/awslabs/data-on-eks)

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

01.repos/terraform-aws-eks-data-addons - (main) > pre-commit run --all-files
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check for merge conflicts................................................Passed
detect private key.......................................................Passed
detect aws credentials...................................................Passed
Terraform fmt............................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
Terraform validate.......................................................Passed

terraform plan was successful

01.repos/terraform-aws-eks-data-addons - (main) > terraform plan    
var.oidc_provider_arn
  The ARN of the cluster OIDC Provider

  Enter a value: arn:aws:iam::090185061218:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/10F979574F7DEB82000945A1A308FCDA

data.aws_partition.current: Reading...
data.aws_region.current: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.current: Read complete after 0s [id=us-west-2]

Changes to Outputs:
  + aws_efa_k8s_device_plugin = []
  + aws_neuron_device_plugin  = []

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

### CODE REVIEW

File: volcano.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
+################################################################################
+# Volcano Scheduler for TorchX
```
Comment (vara-bonthu) [inline volcano.tf:2]: Replace this text with "Volcano Batch Scheduler" 
Its not just used for TorchX as its used with many other frameworks

File: volcano.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+################################################################################
+
+resource "helm_release" "volcano" {
+  count = var.enable_volcano ? 1 : 0
+
+  name                       = try(var.volcano_helm_config["name"], "volcano")
```
Comment (vara-bonthu) [inline volcano.tf:8]: create a local variable for volcano and substitute in `name` ,  `chart`,  `namespace` as default value

```hcl
local {
  name = "volcano
}
```

File: volcano.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  repository                 = try(var.volcano_helm_config["repository"], "https://github.com/volcano-sh/volcano.git/")
+  chart                      = try(var.volcano_helm_config["chart"], "volcano")
+  version                    = try(var.volcano_helm_config["version"], "1.5")
+  timeout                    = try(var.volcano_helm_config["timeout"], 300)
+  values                     = try(var.volcano_helm_config["values"], null)
```
Comment (vara-bonthu) [inline volcano.tf:13]: Do we need to specify any default values for deploying this Helm chart?

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the PR @elygerou 👍🏼 
PR looks good. I left few minor comments. We also need to validate that this add-on deploys successfully on Kubernetes without any error.

File: volcano.tf (hunk 1 above)
Comment (elygerou) [reply volcano.tf]: Done

File: volcano.tf (hunk 2 above)
Comment (elygerou) [reply volcano.tf]: Done also for namespace as I followed this link's naming: https://volcano.sh/en/docs/installation/

File: volcano.tf (hunk 3 above)
Comment (elygerou) [reply volcano.tf:17]: Did not seem so I run it successfully in an EKS cluster from the test folder of this repo had to change the repo to this: "https://volcano-sh.github.io/helm-charts"

And here is my run outcome: 
terraform-aws-eks-data-addons/test - (main) > kubectl get all -n volcano-system
NAME                                      READY   STATUS      RESTARTS   AGE
pod/volcano-admission-84fd7c8df5-msnbj    1/1     Running     0          2m27s
pod/volcano-admission-init-bhpdt          0/1     Completed   0          2m27s
pod/volcano-controllers-6b7f4c89d-k29gw   1/1     Running     0          2m27s
pod/volcano-scheduler-9c4bf7d4d-vpqsm     1/1     Running     0          2m27s

NAME                                TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
service/volcano-admission-service   ClusterIP   172.20.54.247   <none>        443/TCP    2m28s
service/volcano-scheduler-service   ClusterIP   172.20.217.35   <none>        8080/TCP   2m28s

NAME                                  READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/volcano-admission     1/1     1            1           2m28s
deployment.apps/volcano-controllers   1/1     1            1           2m28s
deployment.apps/volcano-scheduler     1/1     1            1           2m28s

NAME                                            DESIRED   CURRENT   READY   AGE
replicaset.apps/volcano-admission-84fd7c8df5    1         1         1       2m29s
replicaset.apps/volcano-controllers-6b7f4c89d   1         1         1       2m29s
replicaset.apps/volcano-scheduler-9c4bf7d4d     1         1         1       2m29s

NAME                               COMPLETIONS   DURATION   AGE
job.batch/volcano-admission-init   1/1           9s         2m30s

and if you compare with the link everything looks the same Step: 2
Use Helm charts to install Volcano:
https://volcano.sh/en/docs/installation/

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Review by vara-bonthu — APPROVED** [review]
LGTM 🚀

**Pull request merged**